src/azure-cli-core/azure/cli/core/tests/test_profile_v2016_06_01.py

```diff
@@ -1053,7 +1053,7 @@ def __init__(self, *args, **kwargs):
     @mock.patch('azure.cli.core._profile.SubscriptionFinder', autospec=True)
     def test_find_subscriptions_in_vm_with_msi_user_assigned_with_object_id(self, mock_subscription_finder, mock_get_client_class,
                                                                             mock_msi_auth):
-        from requests import HTTPError
+        from ..azclierror import AzureResponseError
```

Comment:
Better to use absolute import , according to [PEP 8](https://www.python.org/dev/peps/pep-0008/#imports).

Reply:
In **CLI Automation Full Test**, [a failure](https://dev.azure.com/azure-sdk/public/_build/results?buildId=586019&view=logs&j=771f6cdf-8d98-5d9b-2fb9-d3a6e9554788&t=0a182f1d-9946-580d-467e-4d4ebd6af32d&l=8924) is reported on test `test_find_subscriptions_in_vm_with_msi_user_assigned_with_object_id `.

```
ERROR: test_find_subscriptions_in_vm_with_msi_user_assigned_with_object_id (core.tests.test_profile.TestProfile)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/mock/mock.py", line 1369, in patched
    return func(*newargs, **newkeywargs)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/azure/cli/core/tests/test_profile.py", line 1231, in test_find_subscriptions_in_vm_with_msi_user_assigned_with_object_id
    subscriptions = profile.find_subscriptions_in_vm_with_msi(identity_id=test_object_id)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/azure/cli/core/_profile.py", line 321, in find_subscriptions_in_vm_with_msi
    msi_creds = MSIAuthenticationWrapper(resource=resource, client_id=identity_id)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/mock/mock.py", line 1100, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/mock/mock.py", line 1104, in _mock_call
    return _mock_self._execute_mock_call(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/mock/mock.py", line 1167, in _execute_mock_call
    result = effect(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/azure/cli/core/tests/test_profile.py", line 1209, in __init__
    self.set_token()
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/azure/cli/core/tests/test_profile.py", line 1219, in set_token
    raise AzureResponseError('Failed to connect to MSI. Please make sure MSI is configured correctly.\n'
core.azclierror.AzureResponseError: Failed to connect to MSI. Please make sure MSI is configured correctly.
Get Token request returned http error: 400, reason: Bad Request
```

This is because in the CI, separate packages will be built for `nosetests` causing relative `import` to refer to another `AzureResponseError`.

If we print the module name of `AzureResponseError` before it is used

```py
print(AzureResponseError.__module__)
```

then run the pipeline again, it will show `AzureResponseError` from 2 places.

```
-------------------- >> begin captured stdout << ---------------------
azure.cli.core.azclierror
core.azclierror
```

That's why the `except` clause fails to capture `AzureResponseError` and the exception is thrown.

https://github.com/Azure/azure-cli/blob/5712e3a9b739b65ec6de314237e6ef9f3a47ca7f/src/azure-cli-core/azure/cli/core/_profile.py#L324-L325